src/scss/components/_input.scss

```diff
@@ -0,0 +1,336 @@
+// input 1
+$input-background: #FAFAFA;
+$input-border: #CCC;
+$input-color: #A475E4;
+$input-placeholder-active: #BBB;
+
+.input-wrapper {
+  height: 40px;
+  border: 1px solid $input-border;
+  margin-bottom: 20px;
+  position: relative;
+  border-radius: 2px;
+  max-width: 300px;
+
+  .input-label {
+    position: absolute;
+    left: 50%;
+    top: 50%;
+    transform: translate(-50%, -50%);
+    color: $input-color;
+    transition: all 0.2s;
+    font-weight: 300;
+    pointer-events: none;
+  }
+
+  .input-base {
+    width: 100%;
+    height: 100%;
+    background-color: transparent;
+    border: none;
+    color: $input-color;
+    padding: 15px 10px 0px 5px;
+    font-size: 14px;
+    font-weight: 300;
+
+    &:focus {
+      outline: none;
+    }
+
+    &.filled-input + .input-label {
+      top: 2px;
+      left: 5px;
+      font-size: 12px;
+      transform: translate(0, 0);
+      transition: all 0.2s;
+      font-weight: 300;
+      color: $input-placeholder-active;
+    }
+  }
+}
+
+// input 2
+.inp {
+  position: relative;
+  width: 100%;
+  max-width: 300px;
+}
+.inp .label {
+  position: absolute;
+  top: 16px;
+  left: 0;
+  font-size: 16px;
+  color: #9098a9;
+  font-weight: 500;
+  transform-origin: 0 0;
+  transition: all 0.2s ease;
+}
+.inp .border {
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  height: 2px;
+  width: 100%;
+  background: #07f;
+  transform: scaleX(0);
+  transform-origin: 0 0;
+  transition: all 0.15s ease;
+}
+.inp input {
+  -webkit-appearance: none;
+  width: 100%;
+  border: 0;
+  font-family: inherit;
+  padding: 12px 0;
+  height: 48px;
+  font-size: 16px;
+  font-weight: 500;
+  border-bottom: 2px solid #c8ccd4;
+  background: none;
+  border-radius: 0;
+  color: #223254;
+  transition: all 0.15s ease;
+}
+.inp input:hover {
+  background: rgba(34,50,84,0.03);
+}
+.inp input:not(:placeholder-shown) + span {
+  color: #5a667f;
+  transform: translateY(-26px) scale(0.75);
+}
+.inp input:focus {
+  background: none;
+  outline: none;
+}
+.inp input:focus + span {
+  color: #07f;
+  transform: translateY(-26px) scale(0.75);
+}
+.inp input:focus + span + .border {
+  transform: scaleX(1);
+}
+
+// input 3
+:root {
```

Comment:
I don't like this CSS variables here, i think that is specific for this component